### PR TITLE
Use Variance library to calculate tolerances in Measurement tests

### PR DESCRIPTION
**Pull request by riemanli**
## TL;DR
Using variances instead of fixed tolerance values for checking measurement values. 

### FrequencyDistributionSubject.kt
- Add a `isWithin` function to compare actual and expected using a tolerance map.

### MeasurementConsumerSimulator.kt
- Add a function to compute reach variances.
- Add a function to compute the margin of error of a 99.9% Confidence Interval (CI).
- Add a function to compute tolerances for relative frequency results.
- Check if reach results and frequency results are within 99.9% CI. 

## Why make this change
Previous checks on measurement results were using fixed tolerance values which is hardcoded and not meaningful. Using CIs provides a statistical way to make sure that the noised results fall into a reasonable range.

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311)

**Comment by riemanli** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1311** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1311" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1311?utm_source=stack-comment).

**Review by SanjayVas — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311)* status: all files reviewed, 1 unresolved discussion (waiting on @renjiezh and @riemanli)

___
*[`src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt` line 212 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311#-Nhbdcnf0hjY4riGbSwM:-Nhbdcnf0hjY4riGbSwN:b-ek8c3w) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6b4175a5a36dcf3252bf9101f7f94bd2fb81ae34/src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt#L212)):*
> ```Kotlin
>       )
>     val reachTolerance = computeErrorMargin(reachVariance)
>     assertThat(reachAndFrequencyResult.reach.value.toDouble())
> ```

nit: Rather than converting to `Double`, you can add `isWithin` to `FuzzyLongSubject`.

**Review by riemanli — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311)* status: 2 of 4 files reviewed, all discussions resolved (waiting on @renjiezh and @SanjayVas)

___
*[`src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt` line 212 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311#-Nhbdcnf0hjY4riGbSwM:-NhbmG5d8xK28_77XKEh:bz5py0o) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6b4175a5a36dcf3252bf9101f7f94bd2fb81ae34/src/main/kotlin/org/wfanet/measurement/loadtest/measurementconsumer/MeasurementConsumerSimulator.kt#L212)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

nit: Rather than converting to `Double`, you can add `isWithin` to `FuzzyLongSubject`.
</blockquote></details>

Good idea! Done.

**Review by renjiezh — APPROVED** [review]
> Reviewed 3 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)

**Review by SanjayVas — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Marco-Premier)

**Review by Marco-Premier — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1311)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @riemanli)
